mastodon/src/main/java/org/joinmastodon/android/utils/StatusFilterPredicate.java

```diff
@@ -72,9 +85,21 @@ public boolean test(Status status){
 				// only apply filters for given context
 				.filter(filter -> filter.context.contains(context))
 				// treating filterAction = null (from filters list) as FilterAction.HIDE
-				.filter(filter -> filter.filterAction == null ? action == Filter.FilterAction.HIDE : filter.filterAction == action)
+				.filter(filter -> filter.filterAction == null ? action == HIDE : filter.filterAction == action)
 				.findAny();
 
+		//Apply client filters if no server filter is triggered
+		if (applyingFilter.isEmpty()) {
+			applyingFilter = clientFilters.stream()
+					// only apply filters for given context
+					.filter(filter -> filter.context.contains(context))
+					// treating filterAction = null (from filters list) as FilterAction.HIDE
+					.filter(filter -> filter.filterAction == null ? action == HIDE : filter.filterAction == action)
+					//client filter has to match the status
```

Comment:
Please use comments to add additional context to code. It is fairly obvious from the that the filter has to match the status.

Reply:
I copied the comments from above -> removed them